packages/utils/src/lib/git.integration.test.ts

```diff
@@ -16,3 +24,66 @@ describe('getLatestCommit', () => {
     );
   });
 });
+
+describe('branchHasChanges', () => {
+  it('should return true if some changes are given', async () => {
+    await makeStatusDirty();
+    await expect(branchHasChanges()).resolves.toBe(true);
+    await makeStatusClean();
+  });
+  it('should return false if no changes are given', async () => {
+    await expect(branchHasChanges()).resolves.toBe(false);
+  });
+});
+
+describe('guardAgainstDirtyRepo', () => {
+  it('should throw if history is dirty', async () => {
+    await makeStatusDirty();
+    await expect(guardAgainstDirtyRepo()).rejects.toThrow(
+      'Repository should be clean before we you can proceed',
+    );
+    await makeStatusClean();
+  });
+  it('should not throw if history is clean', async () => {
+    await expect(guardAgainstDirtyRepo()).resolves.toEqual(void 0);
```

Comment:
`void 0` is a strange way to write `undefined` :sweat_smile: 

```suggestion
    await expect(guardAgainstDirtyRepo()).resolves.toBeUndefined();
```